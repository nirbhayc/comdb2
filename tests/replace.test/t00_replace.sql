CREATE TABLE t1(i INT PRIMARY KEY) $$
INSERT INTO t1 VALUES (1), (2);

BEGIN;
INSERT INTO t1 VALUES (2);
INSERT INTO t1 VALUES (3);
COMMIT;
SELECT * FROM t1;

BEGIN;
REPLACE INTO t1 VALUES (2);
INSERT INTO t1 VALUES (3);
COMMIT;
SELECT * FROM t1;

REPLACE INTO t1 VALUES (1), (2), (3), (4);
SELECT * FROM t1;

CREATE TABLE t2(i INT PRIMARY KEY, c CSTRING(100)) $$
INSERT INTO t2 VALUES (1, "aaa"), (2, "bbb");

BEGIN;
REPLACE INTO t2 VALUES (1, "ccc"), (2, "ddd");
INSERT INTO t2 VALUES (1, "ccc"), (2, "ddd");
COMMIT;
SELECT * FROM t2;

REPLACE INTO t2 VALUES (1, "ccc"), (2, "ddd");
SELECT * FROM t2;

DROP TABLE t1;
DROP TABLE t2;

exec procedure sys.cmd.send('verify t1');
