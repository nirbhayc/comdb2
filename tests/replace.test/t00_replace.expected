(rows inserted=2)
[COMMIT] failed with rc 299 add key constraint duplicate key 'COMDB2_PK' on table 't1' index 0
(i=1)
(i=2)
(i=1)
(i=2)
(i=3)
(rows inserted=4)
(i=1)
(i=2)
(i=3)
(i=4)
(rows inserted=2)
[COMMIT] failed with rc 299 add key constraint duplicate key 'COMDB2_PK' on table 't2' index 0
(i=1, c='aaa')
(i=2, c='bbb')
(rows inserted=2)
(i=1, c='ccc')
(i=2, c='ddd')
